fury/actor.py

```diff
@@ -2506,6 +2506,34 @@ def texture(rgb, interp=True):
     return act
 
 
+def change_texture(actor_obj, rgb, interp=True):
+    """ Map an RGB or RGBA texture on an already textured vtkActor
+
+    Parameters
+    ----------
+    actor_obj: vtkActor
+        An already Textured vtkActor
+    rgb : ndarray
+        Input 2D RGB or RGBA array. Dtype should be uint8.
+    interp : bool
+        Interpolate between grid centers. Default True.
+
+    Returns
+    -------
+    vtkActor
+    """
+    grid = rgb_to_vtk(np.ascontiguousarray(rgb))
+
+    # Create texture object
+    texture = vtk.vtkTexture()
+    texture.SetInputDataObject(grid)
+
+    if interp:
+        texture.InterpolateOn()
+
+    actor_obj.SetTexture(texture)
```

Comment:
Your code here is doing a lot of copy and conversion. That's why it is cpu intensive. 
Using [`numpy_to_vtk_colors`](https://github.com/fury-gl/fury/blob/master/fury/utils.py#L53) could help you. in this function, you will have to add another parameter to avoid deepcopy ([line 76](https://github.com/fury-gl/fury/blob/master/fury/utils.py#L76)).
Also, you do not need to create a texture each time. You want to update it. So I suppose that you could reach and replace the previous texture array instead of delete + copy

Reply:
Ok, great thanks for the reply. I will take a look and make the required changes.

Reply:
@skoudoro can we have a bit detailed conversation about this issue. numpy_to_vtk seems to take only 2d vectors I wonder how will it take an image as It throws ```AssertionError: Only arrays of dimensionality 2 or lower are allowed!``` for an RGB image.

Moreover, I tried to look into vtk documentation but did not find a way to change the texture, in some way other than creating one and setting it.

Reply:
This is resolved